dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -165,14 +169,42 @@ public ResponseEntity<String> listHandles(HttpServletRequest request, @PathVaria
     }
 
     /**
-     * Maps the handle to a correct response.
+     * Maps the handle to url.
      *
      * @param request        HttpServletRequest
      * @param handleResolver HdlResolverDTO - Handle resolver
-     * @return One element list using String if found, else null String.
+     * @return String if found, else null String.
      */
     private String resolveToURL(HttpServletRequest request, HdlResolverDTO handleResolver) {
-        return mapAsJson(this.hdlResolverService.resolveToURL(ContextUtil.obtainContext(request), handleResolver));
+        return this.hdlResolverService.resolveToURL(ContextUtil.obtainContext(request), handleResolver);
+    }
+
+    /**
+     * Resolves the metadata based on the given request and handle resolver.
+     *
+     * @param request        the HTTP request containing the metadata parameter
+     * @param handleResolver the handle resolver containing the handle
+     * @return a JSON representation of the URL or the handle values map, or "null" in case of an error
+     */
+    private String resolveToMtd(HttpServletRequest request, HdlResolverDTO handleResolver) {
+        Map<String, String> result = new HashMap<>();
+        String url = resolveToURL(request, handleResolver);
+        String param = request.getParameter("metadata");
+        if (StringUtils.isBlank(param)) {
+            return mapAsJson(url);
+        }
+        result.put("url", url);
+        String handle = handleResolver.getHandle();
+        try {
+            Map<String, String> metadata = HandlePlugin.getMapHandleValues(handle);
+            for (Map.Entry<String, String> entry : metadata.entrySet()) {
+                result.put(entry.getKey().toLowerCase(), entry.getValue());
```

Comment:
why lowercase?

Reply:
I thought using lowercase might make it easier to use in the future, but it actually makes the code harder to read.